src/server/modes/base/responses/afk-disconnect.ts

```diff
@@ -0,0 +1,49 @@
+import { ServerPackets, SERVER_ERRORS, SERVER_PACKETS } from '@airbattle/protocol';
+import { CONNECTIONS_BREAK, CONNECTIONS_SEND_PACKET, ERRORS_AFK_DISCONNECT } from '@/events';
+import { System } from '@/server/system';
+import { MainConnectionId } from '@/types';
+import { CONNECTIONS_STATUS } from '@/constants';
+
+export default class AfkDisconnectResponse extends System {
+  constructor({ app }) {
+    super({ app });
+
+    this.listeners = {
+      [ERRORS_AFK_DISCONNECT]: this.onAfkDisconnect,
+    };
+  }
+
+  /**
+   * Show an error message.
+   *
+   * @param connectionId
+   */
+  onAfkDisconnect(connectionId: MainConnectionId): void {
+    if (!this.storage.connectionList.has(connectionId)) {
+      return;
+    }
+
+    const connection = this.storage.connectionList.get(connectionId);
+
+    if (connection.meta.status !== CONNECTIONS_STATUS.PENDING_TO_CLOSE) {
```

Comment:
If it is done in the same way as a packet flooding connection break, it is necessary there to avoid punishing the player twice for the same infringement (because before the connection is broken the player can continue to send packets). There doesn't seem to be a need for such checks here.

Reply:
I added this because otherwise `onUpdatePlayers` will send `ERROR_AFK_DISCONNECT` for that player on every tick until their connection is broken, so the player receives multiple `AFK_DISCONNECT` errors and has more than one connection break timeout set up. 

Maybe this is the wrong place for it and preventing `onUpdatePlayers` from emitting `ERROR_AFK_DISCONNECT` more than once per AFK player would be a cleaner approach?

Reply:
Yeah, you're right. I'll think of a way to simplify it, this isn't the only place with a problem like this.